pkg/gcrane/gc.go

```diff
@@ -17,62 +17,182 @@ package gcrane
 import (
 	"fmt"
 	"log"
+	"net/http"
+	"time"
 
 	"github.com/google/go-containerregistry/pkg/authn"
 	"github.com/google/go-containerregistry/pkg/name"
 	"github.com/google/go-containerregistry/pkg/v1/google"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
 	"github.com/spf13/cobra"
 )
 
 func init() { Root.AddCommand(NewCmdGc()) }
 
 func NewCmdGc() *cobra.Command {
+	before := ""
+	after := ""
+
 	recursive := false
+	remove := false
+	untag := false
+
 	cmd := &cobra.Command{
 		Use:   "gc",
-		Short: "List images that are not tagged",
+		Short: "Garbage collect images",
 		Args:  cobra.ExactArgs(1),
 		Run: func(_ *cobra.Command, args []string) {
-			gc(args[0], recursive)
+			opts, err := parseOpts(before, after, recursive, remove, untag)
+			if err != nil {
+				log.Fatalln(err)
+			}
+
+			if err := GarbageCollect(args[0], opts); err != nil {
+				log.Fatalln(err)
+			}
 		},
 	}
 
+	// Filters
+	cmd.Flags().StringVarP(&before, "before", "b", "", "Match images uploaded before this time (RFC3339)")
+	cmd.Flags().StringVarP(&after, "after", "a", "", "Match images uploaded after this time (RFC3339)")
+
+	// Behaviors
 	cmd.Flags().BoolVarP(&recursive, "recursive", "r", false, "Whether to recurse through repos")
+	cmd.Flags().BoolVarP(&remove, "delete", "D", false, "Delete images instead of just printing them")
```

Comment:
What do you think about calling this `--dry_run` where true (the default) prints images it _would_ delete, and false would list and then delete them. `-D` could still work for the short name.

Reply:
If it's defaulting to true, then I think you have to do `-D=false` or something for the short name, which is gross. I made these all default to false so that an empty struct for `GCOptions` would work, but adding functional options could fix that... I'll think on it.

Reply:
I like --dry_run as an option name (or --dry-run)